actors/builtin/miner/miner_actor.go

```diff
@@ -1685,6 +1682,12 @@ func handleProvingDeadline(rt Runtime) {
 			pledgeDeltaTotal = big.Add(pledgeDeltaTotal, newlyVested.Neg())
 		}
 
+		{
+			// Process pending worker change if any
+			info := getMinerInfo(rt, &st)
+			processPendingWorker(info, rt, &st)
+		}
+
```

Comment:
At a high level, when will `processPendingWorker` be called by `handleProvingDeadline` rather than `ConfirmUpdateWorkerKey`?

Reply:
It's called by `handleProvingDeadline` which is executed at the end of every deadline (30 mins or 48 times a day). This is the same average frequency we would expect for SubmitPoSt if the miner has sectors in every deadline and sends proofs for all partitions in one message.

Reply:
So when would the `ConfirmUpdateWorkerKey` happen?

Reply:
It's called by `handleProvingDeadline` which is executed at the end of every deadline (30 mins or 48 times a day). This is the same average frequency we would expect for SubmitPoSt if the miner has sectors in every deadline and sends proofs for all partitions in one message.

Reply:
`ConfirmUpdateWorkerKey` can be called by the miner's owner any time, but it will only have an effect when a pending worker key has been set and the current epoch is at or past it's effective epoch. It allows the miner's owner to spend a little gas to update the key as soon as possible.

Reply:
I see! If @wadeAlexC's suggestion of quantizing the `EffectiveAt` time below is applied, then the call should be removed. 

Though again, if there is a security issue (ie compromised keys), it's definitely nice for the miner to have the option to do this.